### PR TITLE
serial refactoring

### PR DESCRIPTION
**Pull request by AndreySmirnov81**
@burrbull

### CODE REVIEW

File: src/serial.rs [hunk 1]
```diff
@@ -320,21 +319,19 @@ where
...
-                .pce()
-                .bit(parity_control_enable)
+        self.usart.cr1.modify(|_r, w| { w
+            .m().bit(match config.wordlength {
+                WordLength::DataBits8 => false,
+                WordLength::DataBits9 => true,
+            })
+            .ps().bit(parity_is_odd)
+            .pce().bit(parity_is_used)
         });
```
Comment (burrbull) [inline src/serial.rs:335]: If you want to write it in 1 line you can use this way
```suggestion
        self.usart.cr1.modify(|_r, w| {
            w.m().bit(match config.wordlength {
                WordLength::DataBits8 => false,
                WordLength::DataBits9 => true,
            });
            w.ps().bit(parity_is_odd);
            w.pce().bit(parity_is_used)
        });
```
or rustfmt

**Comment by burrbull** [comment]
LGTM. Please squash commits

File: src/serial.rs (hunk 1 above)
Comment (AndreySmirnov81) [reply src/serial.rs]: Thanks for the hint

**Pull request merged**